backend/apps/core/services/validation_rules.py

```
@@ -0,0 +1,159 @@
+"""
+Validation rule classes for extensible validation.
+
+Provides pluggable validation rules that can be composed together,
+addressing Open/Closed Principle (OCP) violations.
+"""
+from typing import Dict, Any, List, Optional
```

Ruff (0.14.3)

`typing.Dict` is deprecated, use `dict` instead

(UP035)